src/core/prometheus-impl.hh

```diff
@@ -19,13 +19,76 @@
 
 
 #include <boost/container_hash/hash_fwd.hpp>
+#include <ranges>
 #include <seastar/core/metrics.hh>
 #include <seastar/core/metrics_api.hh>
+#include <string_view>
 
 #include <unordered_map>
 #include <vector>
 #include <string>
 
+namespace seastar::prometheus {
+
+
+namespace {
```

Comment:
Unnamed namespaces in a header are transformed to different external names for each translation unit, causing bloat. Better use `namespace internal`.

Reply:
One interesting note: just dropping the `fmt_but` class into an anonymous namespace, instead of a named one, improved perf signitifantly, like 5% or more over the whole benchmark. Probably the visibility change affects the inlining heuristic? Since the compiler knows there are no other callers outside the current TU it can make a more accurate assessment of inlining bloat.

Reply:
(ignore the 5% loss, not the potential bloat from unnamed namespaces)